src/sage/combinat/bijectionist.py

```diff
@@ -0,0 +1,3230 @@
+            {[]: 0, [1, 0]: 1, [1, 0, 1, 0]: 1, [1, 1, 0, 0]: 2}
+            {[]: 0, [1, 0]: 1, [1, 0, 1, 0]: 2, [1, 1, 0, 0]: 1}
+            sage: for subdistribution in bij.minimal_subdistributions_iterator():
+            ....:     print(subdistribution)
+            ([[]], [0])
+            ([[1, 0]], [1])
+            ([[1, 0, 1, 0], [1, 1, 0, 0]], [1, 2])
+
+        An example with two elements of the same block in a subdistribution::
+
+            sage: A = B = ["a", "b", "c", "d", "e"]
+            sage: tau = {"a": 1, "b": 1, "c": 2, "d": 2, "e": 3}.get
+            sage: bij = Bijectionist(A, B, tau)
+            sage: bij.set_constant_blocks([["a", "b"]])
+            sage: bij.set_value_restrictions(("a", [1, 2]))
+            sage: bij.constant_blocks(optimal=True)
+            {{'a', 'b'}}
+            sage: list(bij.minimal_subdistributions_iterator())
+            [(['a', 'b', 'c', 'd', 'e'], [1, 1, 2, 2, 3])]
+        """
+        # see
+        # https://mathoverflow.net/questions/406751/find-a-subdistribution/406975
+        # and
+        # https://gitlab.com/mantepse/bijection-tools/-/issues/29
+
+        minimal_subdistribution = MixedIntegerLinearProgram(maximization=False, solver=self._solver)
+        D = minimal_subdistribution.new_variable(binary=True)  # the subset of elements
+        V = minimal_subdistribution.new_variable(integer=True)  # the subdistribution
+        minimal_subdistribution.set_objective(sum(D[a] for a in self._A))
+        minimal_subdistribution.add_constraint(sum(D[a] for a in self._A) >= 1)
+
+        if self._bmilp is None:
+            self._bmilp = _BijectionistMILP(self)
+        s = next(self._bmilp.solutions_iterator(False, []))
+        while True:
+            for v in self._Z:
+                minimal_subdistribution.add_constraint(sum(D[a] for a in self._A if s[a] == v) == V[v])
+            try:
+                minimal_subdistribution.solve()
+            except MIPSolverException:
+                return
+            d = minimal_subdistribution.get_values(D, convert=bool, tolerance=0.1)  # a dict from A to {0, 1}
+            new_s = self._find_counter_example(self._A, s, d, False)
+            if new_s is None:
+                values = self._sorter["Z"](s[a] for a in self._A if d[a])
+                yield ([a for a in self._A if d[a]], values)
+
+                # get all variables with value 1
+                active_vars = [D[a] for a in self._A
+                               if minimal_subdistribution.get_values(D[a], convert=bool, tolerance=0.1)]
+
+                # add constraint that not all of these can be 1, thus vetoing
+                # the current solution
+                minimal_subdistribution.add_constraint(sum(active_vars) <= len(active_vars) - 1,
+                                                       name="veto")
+            else:
+                s = new_s
+
+    def _find_counter_example(self, P, s0, d, on_blocks):
```

Comment:
```suggestion
    def _find_counterexample(self, P, s0, d, on_blocks):
```
Typically counterexample is written as one word in English (at least as far as I have seen and recall).